pymeasure/display/windows/multiplot_window.py

```diff
@@ -0,0 +1,132 @@
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+import os
+
+import pyqtgraph as pg
+
+from .managed_window import ManagedWindowBase
+from ..widgets.multiplot_widget import MultiPlotWidget, MultiPlotResultsDialog
+from ..widgets.log_widget import LogWidget
+from ..browser import BrowserItem
+from ..Qt import QtWidgets
+from ..manager import Experiment
+from ...experiment import Results
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class MultiPlotWindow(ManagedWindowBase):
+    """
+    A window for plotting multiple graphs that share the X axis with custom Y axis
+
+    .. seealso::
+        Tutorial :ref:`tutorial-plotterwindow`
+        A tutorial and example code for using the Plotter and PlotterWindow.
+
+    Parameters for :code:`__init__` constructor.
+
+    :param procedure_class: procedure class describing the experiment (see
+        :class:`~pymeasure.experiment.procedure.Procedure`)
+    :param x_axis: the initial data-column for the x-axis of the plot
+    :param y_axis: the initial data-column for the y-axis of the plot, either a single y-axis label
+        or a list of y-axis labels for each plot
+    :param linewidth: linewidth for the displayed curves, default is 1
+    :param num_plots: number of plots to display, default is 1
+    :param \\**kwargs: optional keyword arguments that will be passed to
+        :class:`~pymeasure.display.windows.managed_window.ManagedWindowBase`
+
+    """
+
+    def __init__(self, procedure_class, x_axis=None, y_axis=None, linewidth=1, num_plots=1,
+                 **kwargs):
+        self.x_axis = x_axis
+        self.y_axis = y_axis
+        self.num_plots = num_plots
+        self.log_widget = LogWidget("Experiment Log")
+        self.plot_widget = MultiPlotWidget("Results Graph", procedure_class.DATA_COLUMNS, x_axis,
+                                           y_axis,
+                                           num_plots=num_plots, linewidth=linewidth)
+        self.plot_widget.setMinimumSize(100, 200)
+
+        if "widget_list" not in kwargs:
+            kwargs["widget_list"] = ()
+        kwargs["widget_list"] = kwargs["widget_list"] + (self.plot_widget, self.log_widget)
+
+        super().__init__(procedure_class, **kwargs)
+
+        # Setup measured_quantities once we know x_axis and y_axis
+        measure_quantities = [self.x_axis, self.y_axis]
+        if type(self.y_axis) == list:
```

Comment:
for type checking you can better use `isinstance(self.y_axis, list)`

Reply:
Replaced with isinstance()